seawatch_registration/views/profile.py

```diff
@@ -39,31 +41,47 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
-class DetailView(LoginRequiredMixin, UserPassesTestMixin, generic.DetailView):
+class DetailView(LoginRequiredMixin, HasProfileMixin, generic.DetailView):
     model = Profile
     nav_item = 'profile'
 
     def get_object(self):
         return Profile.objects.get(user=self.request.user)
 
-    def test_func(self):
-        return Profile.objects.filter(user=self.request.user).exists()
 
-
-class UpdateView(LoginRequiredMixin, UserPassesTestMixin, RedirectNextMixin, generic.UpdateView):
+class UpdateView(LoginRequiredMixin, RedirectNextMixin, generic.View):
     navitem = 'profile'
-    model = Profile
-    template_name = './seawatch_registration/profile.html'
-    success_url = reverse_lazy('profile_detail')
     submit_button = 'Save'
-    form_class = ProfileForm
+    template_name = './seawatch_registration/profile_update.html'
+    success_url = reverse_lazy('profile_detail')
+
+    def get(self, request):
```

Comment:
Since all you do here is to fill context you could use the `get_context_data()` method. By including the `super()` context data view is automatically included.